### PR TITLE
Histogram writer for hs00 schema

### PR DESCRIPTION
**Pull request by dominikwerder**
### Description of work

Let filewriter write histogram data from the `hs00` schema.

- Histograms can come from a single or from multiple flatbuffer messages which cover mutually exclusive slices.
- All numeric types as defined in schema `hs00` are accepted.
- `README` describes the available options.

Used (so far) at AMOR at PSI.

### CODE REVIEW

File: src/schemas/hs00/WriterTyped.h [hunk 1]
```diff
@@ -0,0 +1,521 @@
...
+    hdf5::Dimensions DimsOut;
+    hdf5::dataspace::Simple SpaceOut;
+    {
+      DimsOut.resize(DimsMem.size() - 1);
+      for (size_t I = 1; I < DimsMem.size(); ++I) {
+        DimsOut.at(I - 1) = DimsMem.at(I);
+      }
+      SpaceOut = hdf5::dataspace::Simple(DimsOut, DimsOut);
+      /*
+      hdf5::Dimensions ChunkElements(DimsOut.size());
```
Comment (michele-brambilla) [inline src/schemas/hs00/WriterTyped.h]: Can this commented block be removed?

File: src/tests/EventHistogramWriterTests.cpp [hunk 2]
```diff
@@ -0,0 +1,547 @@
...
+
+TEST_F(EventHistogramWriter, SlicePartialOverlap) {
+  auto A = Slice::fromOffsetsSizes({1, 6}, {2, 2});
+  auto B = Slice::fromOffsetsSizes({0, 7}, {2, 1});
+  ASSERT_TRUE(A.doesOverlap(B));
+  ASSERT_TRUE(B.doesOverlap(A));
+}
+
+json createTestWriterTypedJson() {
+  // The option 'convert_edge_type_to_float' is a temporary special feature for
```
Comment (michele-brambilla) [inline src/tests/EventHistogramWriterTests.cpp]: I would rather say that there's this extra conversion feature. Or at least say that is "required" (unless for some reason can/will be removed in future)

File: src/tests/EventHistogramWriterTests.cpp [hunk 3]
```diff
@@ -0,0 +1,547 @@
...
+  Dataspace.selection(hdf5::dataspace::SelectionOperation::SET,
+                      hdf5::dataspace::Hyperslab({1, 0, 0, 0}, {1, 4, 2, 2}));
+  hdf5::dataspace::Simple SpaceMem({4, 2, 2});
+  Histograms.read(Buffer, Histograms.datatype(), SpaceMem, Dataspace);
+  for (size_t Flat = 0; Flat < Buffer.size(); ++Flat) {
+    ASSERT_EQ(Buffer.at(Flat), getValueAtFlatIndex(17, Flat, DimLengths));
+  }
+}
+
+TEST_F(EventHistogramWriter, WriteMultipleHistogramsWithMinimumInterval) {
```
Comment (michele-brambilla) [inline src/tests/EventHistogramWriterTests.cpp]: Can we disable it for the moment? 
`TEST_F(DISABLED_EventHistogramWriter, WriteMultipleHistogramsWithMinimumInterval)` should work.

File: src/tests/EventHistogramWriterTests.cpp (hunk 2 above)
Comment (dominikwerder) [reply src/tests/EventHistogramWriterTests.cpp]: I've created #304 and expanded the comment in the code.

**Pull request merged**